client/host_queue.go

```diff
@@ -318,6 +320,29 @@ func (q *queue) asyncFetch(wg *sync.WaitGroup, op *fetchBatchOp) {
 	}()
 }
 
+func (q *queue) asyncTruncate(wg *sync.WaitGroup, op *truncateOp) {
+	wg.Add(1)
+
+	go func() {
+		client, err := q.connPool.NextClient()
+		if err != nil {
+			// No client available
+			op.completionFn(nil, err)
+			wg.Done()
+			return
+		}
+
+		ctx, _ := thrift.NewContext(q.opts.GetTruncateRequestTimeout())
+		if err := client.TruncateNamespace(ctx, &op.request); err != nil {
+			op.completionFn(nil, err)
+		} else {
+			op.completionFn(nil, nil)
+		}
+
+		wg.Done()
```

Comment:
This should probably be `defer wg.Done()` at the top so it is run during the error case too yeah?